src/CMakeLists.txt

```diff
@@ -32,30 +32,31 @@ add_library(libappimage
 )
 
 set_target_properties(libappimage PROPERTIES PREFIX "")
+set_target_properties(libappimage PROPERTIES PUBLIC_HEADER "${PROJECT_SOURCE_DIR}/include/appimage/libappimage.h")
 
 target_compile_definitions(libappimage
     PRIVATE -D_FILE_OFFSET_BITS=64
     PRIVATE -DVERSION_NUMBER="${GIT_VERSION}"
     PRIVATE -DENABLE_BINRELOC
 )
 
-target_include_directories(libappimage PUBLIC
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
-    $<INSTALL_INTERFACE:include>
-)
+target_include_directories(libappimage
+    PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include/>
+    INTERFACE $<INSTALL_INTERFACE:include/> )
 
 target_link_libraries(libappimage
-    pthread
+    PRIVATE
```

Comment:
Good idea to separate between `PRIVATE` and `PUBLIC` linkage here. I hope `PRIVATE` ones are hidden automatically by CMake in the linkage interface.

TODO: verify this assumption

Reply:
Update: Doesn't seem to have any effect, just checked with `nm -g`. But it makes sense even if it's just informational.